### PR TITLE
chore(dmaasbackup): basic controller for dmaasbackup

### PR DESCRIPTION
**Pull request by mynktl**
Changes:
- Added support for controller in server
- Added basic controller for dmaasbackup

### CODE REVIEW

File: pkg/controller/base_controller.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+	qObj, ok := key.(workQueueLoad)
+	if !ok {
+		c.workQueue.Forget(key)
+	}
+
+	var err error
+
+	switch qObj.operation {
+	case qOpDestroy:
```
Comment (vishnuitta) [inline pkg/controller/base_controller.go]: why this separate handling from Sync?
if I'm not wrong, if a CR is deleted when the controller is down, there are chances that qOpDestroy never comes to controller

File: pkg/cmd/server.go [hunk 2]
```diff
@@ -120,9 +170,119 @@ func newLogger(opts *serverOpts) *logrus.Logger {
...
+		opts:                  opts,
+	}, nil
+}
+
+var (
+	// defaultControllerWorker is default value of controller worker
+	defaultControllerWorker int = 1
+)
+
+// listController returns list of supported controller
```
Comment (vishnuitta) [inline pkg/cmd/server.go]: listController sounds like 'get the existing ones'.. but here new ones are created..
can we change to 'getSupportedControllers'?

Comment (mynktl) [reply pkg/cmd/server.go]: ok

File: pkg/controller/base_controller.go (hunk 1 above)
Comment (mynktl) [reply pkg/controller/base_controller.go]: yes. This one is to handle when controller is up. Adding finalizers, which backupcontroller removes, will make sure that this event gets generated.

**Review by vishnuitta — APPROVED** [review]
changes are good

**Pull request merged**